### ISSUE TITLE
[Blog] Error and success state focus not well managed

### ISSUE THREAD

**Issue by svinkle** [Shopify/dawn#696]
## Issue summary

When the blog post comment form is submitted, either with valid or invalid data, a hash is added to the URL. Focus is meant to shift to the error or success heading on load, however the heading does not receive keyboard focus. This forces the user to navigate from the top of the document.

<details>
  <summary>Screenshots</summary>

  <img width="355" alt="Dawn blog post page, error state heading is highlighted." src="https://user-images.githubusercontent.com/1392632/135155704-6d40aea5-a79c-4d34-90c6-ea1cead50a97.png">

</details>

#### HTML

```html
<form
  method="post"
  action="/blogs/news/welcome-home-hem-founder-petrus-palmer/comments#comment_form"
  id="comment_form"
  accept-charset="UTF-8"
  class="comment-form"
>
  <!-- … -->
  <h3 class="form-status caption-large" tabindex="-1" autofocus>
    <!-- Icon… -->
    Please adjust the following:
  </h3>
  <!-- … -->
</form>
```

## Steps to reproduce

1. Go to a Dawn theme store with any supported browser
2. Navigate to a blog list page
3. Activate a blog list link with comments
4. With an active screen reader, use the virtual cursor to navigate to the comment form and submit with invalid data

### Expected

- Keyboard focus to shift to the error/success heading.

### Actual

- Keyboard focus remains at the top of the document.

## Recommendation

Remove the URL hash portion from the forme `action` attribute. This will allow the `autofocus` attribute to function as expected and shift keyboard focus to the appropriate heading.

#### HTML

```html
<form
  method="post"
  action="/blogs/news/welcome-home-hem-founder-petrus-palmer/comments"
  id="comment_form"
  accept-charset="UTF-8"
  class="comment-form"
>
  <!-- … -->
</form>
```

<!-- #### Diff

```diff
- line removed
+ line added
``` -->

## Specifications

- **Browser:** Any
- **Dawn version:** 2.1.0
- **Component:** Blog
- **WCAG Principal:** Operable
- **WCAG SC:** [2.1.1 Keyboard](https://www.w3.org/WAI/WCAG21/Understanding/keyboard.html)
- **Severity:** Medium
- **Effort:** Low
